src/MSALWrapper.Test/ExceptionListToStringConverterTest.cs

```diff
@@ -224,5 +224,37 @@ public void ExtractCorrelationIDs_FromMSALExceptionList_WithOneNullCorrelationID
             // Assert
             correlationIDs.Should().BeEquivalentTo(expectedCorrelationIDs);
         }
+
+        [Test]
+        public void ExceptionList_WithInnerExceptions()
+        {
+            List<Exception> exceptionList = new ()
+            {
+                new Exception("This is the first exception"),
+                new MsalServiceException("2", "This is the second exception", new MsalClientException("3", "This is the inner exception of the second exception")),
+            };
+            var exceptionString = ExceptionListToStringConverter.Execute(exceptionList);
+
+            exceptionString.Should().Be("System.Exception: This is the first exception\n" +
+                "Microsoft.Identity.Client.MsalServiceException: This is the second exception\n" +
+                "Microsoft.Identity.Client.MsalClientException: This is the inner exception of the second exception");

```

Comment:
The expectation right now is that each top level exception we catch is on it's own line. So with this format we would lose the connection that an inner exception was actually an inner exception of the previous, and not a new top level exception we caught. Does that make sense? Perhaps we should move to formatting the exception lists to JSON so that inner exceptions can be recursively nested as needed?

Reply:
Yes Kyle! I thought about this scenario and in my head, I justified that the inner exceptions come right after the original exception, but you are right! I will try formatting the exceptions to a JSON array/list and see how it goes.

Reply:
I just remembered why we didn't do it before. Earlier we thought of serializing the exceptions, but serialization of 'System.Type' instances is not supported right now and exceptions have certain fields which are 'System.Type' instances. 

So, I will only extract messages from these exceptions and not anything else while serializing.

Reply:
Sample exception when serializing unsupported exception fields:
`System.NotSupportedException: 'Serialization and deserialization of 'System.Type' instances are not supported and should be avoided since they can lead to security issues. Path: $.TargetSite.DeclaringType.'`

Reply:
Nvm, I tried Newtonsoft Json library and that doesn't seem to have any issues.